core/vcpu.c

```diff
@@ -1132,17 +1132,35 @@ void vcpu_save_host_state(struct vcpu_t *vcpu)
     save_host_msr(vcpu);
 
     hstate->hcr2 = get_cr2();
+    hstate->dr0 = get_dr0();
```

Comment:
> Most of the time, the vCPU is not in debugging mode [...], so we don't need to save/load debug registers

I'm curious about this: What happens during a VM-exit event if the guest modifies *dr{0,1,2,3,6}*? Unless "DR exiting" is enabled and we prevent the *drN* registers from being written, this would have side-effects on the host, right?

So saving/loading *drN* registers should occur if **any** of these events occur:
1. "DR exiting" is disabled.
2. "HAX_DEBUG_USE_HW_BP" is enabled.

What do you think?

Reply:
That's indeed a very good point. Should we always enable "DR exiting" though, because:

1. If the debugger runs on the host, we can prevent the guest from tampering with DRs.
2. If the debugger runs on the guest, we can keep track of the "dirty" DRs we should save/restore. This is more efficient than having "DR exiting" disabled and saving/restore all the DRs at every guest/host context switch.

If you agree with this approach, could you implement it in this PR (maybe as a separate commit)?